theory.md

```diff
@@ -1,52 +1,63 @@
 ### 1. Regular Expressions
+
 Find regexes that match the following. (e.g. find a single regex that matches
 both `antelope` and `antelopes`.)
 
 a. Single regex that matches either of these:
 
     antelope rocks out
-    
+
     antelopes rock out
 
+regex to match antelope and antelopes: `/antelopes?/`
+regex to match antelope rocks out and antelopes rock out: `/antelopes? rocks? out/`
+
 b. Regex that matches either of:
 
     goat
-    
+
     moat
 
-  but not:
+but not:
 
     boat
 
+regex to match goat and moat: `/[gm]oat/`
+
 c. Regex that matches dates in YYYY-MM-DD format. (Year can be 1-4 digits, and
-  month and day can each be 1-2 digits). This does not need to verify the date
-  is correct (e.g 3333-33-33 can match).
+month and day can each be 1-2 digits). This does not need to verify the date
+is correct (e.g 3333-33-33 can match).
 
     2000-10-12
-  
+
     1999-1-20
-  
+
     1999-01-20
-  
+
     812-2-10
 
+regex to match YYYY-MM-DD: `/[0-9]{4}-[0-9]{1,2}-[0-9]{1,2}/`
```

Comment:
this fails for `812-2-10` . `years can be 1 to 4 digits long` you need to do something like - `/[0-9]{1,4}-[0-9]{1,2}-[0-9]{1,2}/`